Add documentation on using remote URIs for runtime environments (#20352)
ray-project/ray: doc/source/dependency-management.rst

   :language: python
   :start-after: __runtime_env_conda_def_start__
   :end-before: __runtime_env_conda_def_end__

.. _remote-uris:

Remote URIs
-----------

The ``working_dir`` and ``py_modules`` arguments in the ``runtime_env`` dictionary can specify either local path(s) or remote URI(s).

A local path must be a directory path. The directory's contents will be directly accessed as the ``working_dir`` or a ``py_module``.
A remote URI must be a link directly to a zip file. **The zip file must contain only a single top-level directory.**
The contents of this directory will be directly accessed as the ``working_dir`` or a ``py_module``.

For example, suppose you want to use the contents in your local ``/some_path/example_dir`` directory as your ``working_dir``.
If you want to specify this directory as a local path, your ``runtime_env`` dictionary should contain:

.. code-block:: python

    runtime_env = {..., "working_dir": "/some_path/example_dir", ...}

> Suppose instead you want to host its contents remotely and provide a remote URI.

Reviewer: ```suggestion
Suppose instead you want to host your files remotely and provide a remote URI.
```

You would need to first compress the ``example_dir`` directory into a zip file.

> There should be no other files or directories at the top level of the zip file, other than ``example_dir``.

Reviewer: Is it possible to give a quick example of how to do this, like an example command or something?

In general, the zip file's name and the top-level directory's name can be anything.
The top-level directory's contents will be used as the ``working_dir`` (or ``py_module``).
Suppose you upload the compressed ``example_dir`` directory to AWS S3 at the S3 URI ``s3://example_bucket/example.zip``.
Your ``runtime_env`` dictionary should contain:

.. code-block:: python

    runtime_env = {..., "working_dir": "s3://example_bucket/example.zip", ...}

.. warning::

>   Check for hidden files in zipped dependencies.

Reviewer: Can you give a brief explanation or example for how to do this?
